ro_help/hub/forms.py

```diff
@@ -46,6 +47,34 @@ class Meta:
             "statute": AdminResubmitFileWidget,
         }
 
+    def clean_avatar(self):
```

Comment:
Form validation can be helpful, but maybe we can try to validate the size at model level, since it can be used in admin as well.

Since `avatar` is an `ImageField`,  we can have something like this (source from https://stackoverflow.com/questions/54976444/django-form-imagefield-validation-for-certain-width-and-height)

```python

def validate_image_size(max_height: int, max_width: int):
    return def validate(image):
        height = image.file.height 
        width = image.file.width

        if width > max_width or height > max_height:
            raise ValidationError(_("Height or width is larger than what is allowed")) # we need to translate the strings
```

```python
avatar = models.ImageField(_("Avatar"), max_length=300,
                           validators=[validate_image_size(settings.MAX_AVATAR_HEIGHT, settings.MAX_AVATAR_WIDTH)])
```

We can have `validate_image_size` as a shared validator, maybe in a `validation.py` file.

The content-type should be validated by the `ImageField` itself, right? I don't think that we need a content-type validator.

Also, maybe we can use something for cropping as well https://pypi.org/project/django-image-cropping/0.3.7/, so users can directly solve the upload problem in just one step.

Reply:
Seems legit I will do the changes